cli/azd/pkg/project/project_utils.go

```diff
@@ -9,34 +9,36 @@ import (
 	"path/filepath"
 	"time"
 
+	"github.com/azure/azure-dev/cli/azd/pkg/dotignore"
 	"github.com/azure/azure-dev/cli/azd/pkg/rzip"
 	"github.com/otiai10/copy"
 )
 
-// CreateDeployableZip creates a zip file of a folder, recursively.
+// createDeployableZip creates a zip file of a folder, recursively.
 // Returns the path to the created zip file or an error if it fails.
 func createDeployableZip(projectName string, appName string, path string) (string, error) {
-	// TODO: should probably avoid picking up files that weren't meant to be deployed (ie, local .env files, etc..)
+	// Create the output zip file path
 	filePath := filepath.Join(os.TempDir(), fmt.Sprintf("%s-%s-azddeploy-%d.zip", projectName, appName, time.Now().Unix()))
 	zipFile, err := os.Create(filePath)
 	if err != nil {
 		return "", fmt.Errorf("failed when creating zip package to deploy %s: %w", appName, err)
 	}
 
-	if err := rzip.CreateFromDirectory(path, zipFile); err != nil {
-		// if we fail here just do our best to close things out and cleanup
+	// Zip the directory without any exclusions (they've already been handled in buildForZip)
+	err = rzip.CreateFromDirectory(path, zipFile)
+	if err != nil {
 		zipFile.Close()
 		os.Remove(zipFile.Name())
 		return "", err
 	}
 
+	// Close the zip file and return the path
 	if err := zipFile.Close(); err != nil {
-		// may fail but, again, we'll do our best to cleanup here.
 		os.Remove(zipFile.Name())
 		return "", err
 	}
 
-	return zipFile.Name(), nil
+	return filePath, nil
```

Comment:
I'm assuming `filePath` gives us the full absolute path while `zipFile.Name()` was only giving us the filename?  I would just double check that this is honored downstream of the caller.

Reply:
I have confirmed that both filePath and zipFile.Name() return the same value:

![image](https://github.com/user-attachments/assets/4a4b2cab-1908-4cff-8fa8-c2d661b76bb1)